From 7c565b45fd51c222c5698c2cc754ee393e97cb3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 1 Oct 2019 11:21:24 +0200
Subject: [PATCH 6/8] winex11.drv: Restore pointer grab on FocusIn events.

This reverts commit 92177b0b161e91f1d609615d89d8e3199feea33f.

We introduced unnecessary complexity by adding the last_clip_* state,
we can instead use the ClipCursor state.

This restores the ClipCursor on FocusIn events by sending a
WM_X11DRV_CLIP_CURSOR message to the foreground window, which will query
the current clipping rect from the server and apply it.
---
 dlls/winex11.drv/event.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/winex11.drv/event.c b/dlls/winex11.drv/event.c
index 256a22c5f0a..10513dd13e6 100644
--- a/dlls/winex11.drv/event.c
+++ b/dlls/winex11.drv/event.c
@@ -872,6 +872,9 @@ static BOOL X11DRV_FocusIn( HWND hwnd, XEvent *xev )
         break;
     }
 
+    /* ask the desktop window to re-apply the current ClipCursor rect */
+    SendMessageW( GetDesktopWindow(), WM_X11DRV_DESKTOP_CLIP_CURSOR, FALSE, FALSE );
+
     /* ignore wm specific NotifyUngrab / NotifyGrab events w.r.t focus */
     if (event->mode == NotifyGrab || event->mode == NotifyUngrab) return FALSE;
 
-- 
2.35.1

